packages/server/postgres/queries/decrementFreeTemplatesRemaining.ts

```diff
@@ -1,13 +1,27 @@
-import getPg from '../getPg'
-import {decrementFreeRetroTemplatesRemainingQuery} from './generated/decrementFreeCustomRetroTemplatesRemainingQuery'
-import {decrementFreePokerTemplatesRemainingQuery} from './generated/decrementFreeCustomPokerTemplatesRemainingQuery'
+import getKysely from '../getKysely'
 
 const decrementFreeTemplatesRemaining = async (userId: string, templateType: 'retro' | 'poker') => {
-  const res =
+  const pg = getKysely()
+  const customTemplateType =
     templateType === 'retro'
-      ? await decrementFreeRetroTemplatesRemainingQuery.run({userId}, getPg())
-      : await decrementFreePokerTemplatesRemainingQuery.run({userId}, getPg())
-  return res
+      ? 'freeCustomRetroTemplatesRemaining'
+      : 'freeCustomPokerTemplatesRemaining'
+
+  const userBeforeUpdate = await pg
```

Comment:
-1 this is slightly worse than the SQL you had because if this function gets called twice then we risk the first query being a stale read. If there's ever a way to do something in a single SQL command, that should be the goal!

See [Playground Link](https://old.kyse.link/?p=b&i=haFkqHBvc3RncmVzoXamMC4yNy4yoXPaAaJpbXBvcnQgeyBHZW5lcmF0ZWQgfSBmcm9tICdreXNlbHknCgpkZWNsYXJlIGdsb2JhbCB7CiAgaW50ZXJmYWNlIERCIHsKICAgIHBlcnNvbjogUGVyc29uVGFibGUKICAgIHBldDogUGV0VGFibGUKICB9CgogIGludGVyZmFjZSBQZXJzb25UYWJsZSB7CiAgICBpZDogR2VuZXJhdGVkPHN0cmluZz4KICAgIGZpcnN0X25hbWU6IHN0cmluZwogICAgbGFzdF9uYW1lOiBzdHJpbmcgfCBudWxsCiAgICBjcmVhdGVkX2F0OiBHZW5lcmF0ZWQ8RGF0ZT4KICAgIGFnZTogbnVtYmVyCiAgfQoKICBpbnRlcmZhY2UgUGV0VGFibGUgewogICAgaWQ6IEdlbmVyYXRlZDxzdHJpbmc-CiAgICBuYW1lOiBzdHJpbmcKICAgIG93bmVyX2lkOiBzdHJpbmcKICAgIHNwZWNpZXM6ICdjYXQnIHwgJ2RvZycKICAgIGlzX2Zhdm9yaXRlOiBib29sZWFuCiAgfQp9oXHZfmNvbnN0IHBlcnNvbiA9IGF3YWl0IGRiCiAgLnNlbGVjdEZyb20oJ3BlcnNvbicpCiAgLnNlbGVjdChbJ2lkJywgJ2ZpcnN0X25hbWUnXSkKICAud2hlcmUoJ2lkJywgJz0nLCAnMScpCiAgLmV4ZWN1dGVUYWtlRmlyc3QoKaFjwg==)


```ts
const person = await db
  .updateTable('User')
  .set((eb) => ({[field]: eb(field, '-', '1')} ))
  .where('id', '=', userId)
  .where(field, '>', 0)
  .executeTakeFirst() 
```

Reply:
Makes sense, thanks! Updated